lib/providers/gcm.js

```diff
@@ -70,6 +70,7 @@ GcmProvider.prototype._createMessage = function(notification) {
   Object.keys(notification).forEach(function (key) {
     if (notification[key] !== null && typeof notification[key] !== 'undefined') {
       message.addData(key, notification[key]);
+      message.addNotification(key, notification[key]);
```

Comment:
The notification field is used for the new gcm apn interface.
I assumed that the notification should be the same for both android and ios.

On Mon, Aug 3, 2015, 17:56 Raymond Feng <notifications@github.com> wrote:

> In lib/providers/gcm.js
> <https://github.com/strongloop/loopback-component-push/pull/98#discussion_r36093048>
> :
>
> > @@ -70,6 +70,7 @@ GcmProvider.prototype._createMessage = function(notification) {
> >    Object.keys(notification).forEach(function (key) {
> >      if (notification[key] !== null && typeof notification[key] !== 'undefined') {
> >        message.addData(key, notification[key]);
> > +      message.addNotification(key, notification[key]);
>
> Do we need somehow distinguish data vs notification?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/strongloop/loopback-component-push/pull/98/files#r36093048>
> .
>